### ISSUE TITLE
Latest seem to break

### ISSUE THREAD

**Issue by arnklint** [drone-plugins/drone-ecr#22]
All steps before this (building etc) works fine. The issue came clear when the latest (looks like #15 might be relevant) image was pushed to docker hub.

```
Drone AWS ECR Plugin built from 55a99ab
$ /usr/bin/docker -d
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
Login failed.
[info] build failed (exit code 1)
```

Tried removing and pulling the `plugins/drone-ecr` image again, the issue remains.

**Comment by joe1chen** [comment]
I am seeing the same issue here as well.  Build `9e49f71` seems to work but `55a99ab` is broken.

```
Drone AWS ECR Plugin built from 55a99ab
$ /usr/bin/docker -d -s overlay
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
```

Is there any way to work around this issue?  I see that on Docker Hub, there are no other tags except `latest`.  So there's no way to rollback to a previous version?  

Currently blocked on this issue, so we'd appreciate if this was resolved asap.

**Comment by edgurgel** [comment]
I'm getting the same issue. Some kind of tagging for plugins would be super useful now.

**Comment by rasharab** [comment]
Same here.
The biggest problem is, as another poster mentioned, there are no historical tags.
And I'm having a hell of time attempting to build the image on my linux box.

I've noticed that you have a commit designed to address this, but it obviously hasn't been deployed.
https://github.com/drone-plugins/drone-ecr/commit/7ac3811094f34b97f60770f4e525b675a21e9e76

Can we please get this fixed as soon as possible, thanks.

-Rashin

**Comment by joe1chen** [comment]
@tboerger Any chance you can help get this fixed?  Or please help with instructions on how to build this by hand?

I tried to clone the repository and updated the `.drone.yml`, added my own `secrets.yml` in an attempt to get this buildling with my own drone 0.4 server.

It failed with the output (debug is turned on):

```
[debug] Successfully decrypted secrets
[debug] Injected secrets into Yaml
[debug] Using workspace /drone/src/bitbucket.org/joe1chen/drone-ecr
[debug] Running Cache step
[debug] Running Clone step
[info] Pulling image plugins/drone-git:latest
Drone Git Plugin built from 43dcd64
$ git init
Initialized empty Git repository in /drone/src/bitbucket.org/joe1chen/drone-ecr/.git/
$ git remote add origin https://bitbucket.org/joe1chen/drone-ecr.git
$ git fetch --no-tags origin +refs/heads/integration:
From https://bitbucket.org/joe1chen/drone-ecr
 * branch            integration -> FETCH_HEAD
 * [new branch]      integration -> origin/integration
$ git reset --hard -q cfb01f089225dae47ce5945c4f6db32547e10e8d
[debug] Running Build and Compose steps
$ make deps
go get -t ./...
package github.com/drone/drone-go/plugin: cannot find package "github.com/drone/drone-go/plugin" in any of:
    /usr/local/go/src/github.com/drone/drone-go/plugin (from $GOROOT)
    /drone/src/github.com/drone/drone-go/plugin (from $GOPATH)
Makefile:16: recipe for target 'deps' failed
make: *** [deps] Error 1
[debug] Running post-Build Cache steps
[info] build failed (exit code 2)
```

This led me to https://github.com/drone/drone-go/issues/27, in which @tboerger mentioned vendoring and pinning to https://github.com/drone/drone-go/commit/eaa41f7836a191224ec5702d2458db07882ec269

Not being a go user, can you explain exactly what needs to be changed to pin drone-go to this specific version?

**Comment by rasharab** [comment]
I'm getting the same issue as the above poster in attempting to build the image and have the same question. :)

-Rashin

**Comment by arnklint** [comment]
Just wanted to highlight the severity of this issue. 
Guess we'll build our own based on this repo; http://readme.drone.io/devs/plugins/#custom-plugins:dce8ed91d073f65a191aa58c2338afcb
and hope for the best.

**Comment by tboerger** [comment]
This plugin have been updated with https://github.com/drone-plugins/drone-ecr/pull/15 and https://github.com/drone-plugins/drone-ecr/pull/14 but even with the CI check there came another issue in. Waiting for a merge of https://github.com/drone-plugins/drone-ecr/pull/23 that hopefully works fine now.

**Comment by arnklint** [comment]
Thanks @tboerger , please let me know if there is anything I can help with here. Noticed a bunch of forks popping up on docker hub since friday, which says something about the usage of this project.

**Comment by tboerger** [comment]
@arnklint can you give the latest version a try?

Beside that we should add more tags in the future to make it possible to roll back and add more automated testing to avoid such issues in the future.

**Comment by arnklint** [comment]
@tboerger works now, thanks!

**Comment by tboerger** [comment]
:+1:

**Comment by joe1chen** [comment]
FWIW - I was finally able to get this building on Drone 0.4 late last night before @tboerger fixed things.

**TLDR:**
The major reason for the failure to compile is due to incompatibilities between Drone 0.4 and the unreleased Drone 0.5.  

There really needs to be some tagging in this source repository to support building with Drone 0.4 and Drone 0.5.  Drone 0.5 is not even released yet, so having this repository configured to build with 0.5 and no branches or tags for 0.4 makes things really hard for everyone in production with 0.4 that want to build this plugin.

**The Long Story:**
For the benefit of everyone who is trying to build this from scratch on Drone 0.4, but couldn't here are the issues I encountered and how I resolved them:

1) The `cannot find package "github.com/drone/drone-go/plugin" in any of:` error seems to be caused by an incompatible version of the `github.com/drone/drone-go/plugin` package being used during compilation.  It seems like the version being used is only compatible with drone 0.5, which is why @tboerger suggested using Go vendoring to pin to a version of the package that is compatible with 0.4.

To add vendoring, you need to either use Go 1.5 with the environment variable `GO15VENDOREXPERIMENT=1` or use Go 1.6.

```
build:
  image: golang:1.5
  environment:
    - GO15VENDOREXPERIMENT=1
    - CGO_ENABLED=0
  commands:
    - make deps
    - make vet
    - make build
    - make test
```

2) Then I added the `vendor.yml` and `vendor` files from https://github.com/drone-plugins/drone-ecr/commit/d4e1d4abfb05dd5fddc552000a3ea7795effe134.  The `drone-go` package is now pinned to use rev `eaa41f7836a191224ec5702d2458db07882ec269`.

```
vendors:
- path: github.com/aws/aws-sdk-go
  rev: d3b6d25135b4e09ecd39ea3c2a5cad35279b0515
- path: github.com/drone/drone-go
  rev: eaa41f7836a191224ec5702d2458db07882ec269
...
```

3) Since I am using Drone 0.4 and not the unreleased 0.5, I did NOT pull in the changes from https://github.com/drone-plugins/drone-ecr/commit/91f0940c9c313d2f925e486df43047a02dce6d8d, which are for Drone 0.5.  (May I suggest an alternate branch with this commit reverted for those of us building with 0.4?)

4) This still did not get the project compiling.  Next I ran into a `use of internal package not allowed` error when pulling dependencies.  These errors were from the `google.golang.org/appengine` package, which didn't seem to be used by this plugin, but were in the `vendor.yml` and `vendor` files.  So I removed references to this package from `vendor.yml` and deleted the files in the `vendor` directory.  Not sure why they were there, but it didn't seem to be using these files.

5) Finally, ran into the same bug as https://github.com/drone-plugins/drone-ecr/pull/23.  The `bytes` package is being imported, but there is a local variable also named `bytes`.  So I renamed the variable to something else.  This finally got everything compiling.

6) To publish to my own repository on Docker Hub, I updated the build settings in `.drone.yml`, created  a new `secrets.yml` with my Docker Hub credentials, and then updated my `.drone.sec` file by running `drone secure --repo joe1chen/drone-ecr --in secrets.yml`.  This is for Drone 0.4, since in 0.5 they've moved to a new mechanism for storing secrets.

**Comment by tboerger** [comment]
@joe1chen you just missed https://github.com/drone-plugins/drone-ecr/tree/release/0.4 this is the source that publishs drone-ecr correctly for drone 0.4.

The master branch will be updated pretty soon to work only with drone >= 0.5

**Comment by joe1chen** [comment]
@tboerger No, actually I saw that branch, which is where I grabbed the vendor assets from.  Please correct me if I'm wrong but the release 0.4 branch contains the change https://github.com/drone-plugins/drone-ecr/commit/91f0940c9c313d2f925e486df43047a02dce6d8d which is titled "Switched to 0.5 build process".  The addition of `.drone.yml.sig` and the removal of `.drone.sec` seems to imply that this change is for Drone 0.5 despite the name of the branch.  Is this check-in a mistake then?

**Comment by tboerger** [comment]
> @tboerger No, actually I saw that branch, which is where I grabbed the vendor assets from. Please correct me if I'm wrong but the release 0.4 branch contains the change 91f0940 which is titled "Switched to 0.5 build process". The addition of .drone.yml.sig and the removal of .drone.sec seems to imply that this change is for Drone 0.5 despite the name of the branch. Is this check-in a mistake then?

This commit just switches to use 0.5 for building the plugin since http://beta.drone.io is running 0.5. without that change we are not able to build the plugin automatically anymore.

Edit: This branch already includes the fix with the variable name collision.

**Comment by joe1chen** [comment]
@tboerger Ok, I see where the confusion is coming from.  The release 0.4 branch is for building a Drone 0.4 compatible version of the plugin using Drone 0.5 (beta.drone.io).

What I'm interested in is a branch that can be used for building a Drone 0.4 compatible version of the plugin using Drone 0.4.  Once again, most people are not using Drone 0.5, so it would be greatly beneficial to have another branch for building with the stable version of Drone.

**Comment by joe1chen** [comment]
As it stands nobody else can build a custom version of this plugin using the master branch or with the 0.4 release branch unless they make significant changes to the source code AND are running their own Drone 0.5.  

And note that just fixing the variable name collision change alone does not get things building with 0.4.

**Comment by tboerger** [comment]
> As it stands nobody else can build a custom version of this plugin using the master branch or with the 0.4 release branch unless they make significant changes to the source code AND are running their own Drone 0.5.

Generally nobody needs to build it with his own drone server because the plugins are getting automatically published. And even if there is an issue that just run through the CI somehow there is still `make docker` to build it locally without the need of any drone server.

> And note that just fixing the variable name collision change alone does not get things building with 0.4.

That's simply not true. The `release/0.4` branch got all required fixes including the vendoring to get it working and buildable.

If you think that the documentation is not clear enough just propose a pull request that improves it. Otherwise simply ask on the Gitter chat or on the discuss forum, that's where you get pretty fast useable answers. or hints how to get something working. There is not more we can provide.

**Comment by joe1chen** [comment]
> Generally nobody needs to build it with his own drone server because the plugins are getting automatically published.

Agreed that nobody wants to build this manually in general.   However, when you are using Drone in production and all releases from your CI server are broken for 4 days, then it becomes critical to be able to at least resolve the problem manually or work around the issue.

> And even if there is an issue that just run through the CI somehow there is still make docker to build it locally without the need of any drone server.

Note that it is **non-trivial** to get this building using `make docker`.  Just pulling down the source code and running `make deps docker` just does not work, at least in my dev environment.  Going through the exercise of building this, it matters greatly what version of Go you are using, what you have $GOPATH set to, etc.  None of these settings are documented anywhere.  I have yet to get this building locally, and I suspect this is the case for the majority of users based on the comments above in this thread.

> That's simply not true. The release/0.4 branch got all required fixes including the vendoring to get it working and buildable.

No, it is true.  Have you tried building the release/0.4 branch with a Drone 0.4 server?  I just tried to build this on my Drone 0.4 server and it does not build using the current `.drone.yml` which has been converted over to the 0.5 format.  

Build Output on Drone 0.4 server with release/0.4 branch:

```
[info] Pulling image plugins/drone-git:latest
Drone Git Plugin built from 43dcd64
$ git init
Initialized empty Git repository in /drone/src/bitbucket.org/joe1chen/drone-ecr/.git/
$ git remote add origin https://bitbucket.org/joe1chen/drone-ecr.git
$ git fetch --no-tags origin +refs/heads/release/0.4:
From https://bitbucket.org/joe1chen/drone-ecr
 * branch            release/0.4 -> FETCH_HEAD
 * [new branch]      release/0.4 -> origin/release/0.4
$ git reset --hard -q 7d456c14f8b503ec5be394ed446432036f5d4698
```

Changing the `.drone.yml` back to 0.4 format does get it building with a Drone 0.4 server, but as it stands today, it does not build with 0.4 server.

> If you think that the documentation is not clear enough just propose a pull request that improves it. Otherwise simply ask on the Gitter chat or on the discuss forum, that's where you get pretty fast useable answers. or hints how to get something working. 

The reason I'm going through the trouble of responding to this at all is for the benefit of future readers who may run into issues building this manually.  

I do agree that the documentation is not clear enough.  

Thanks for the info about the Gitter chat and forum.  Had I known about those resources, I would have taken advantage of them sooner.  

> There is not more we can provide.

I do realize that asking you to provide a `.drone.yml` compatible with a Drone 0.4 server is not really something you'd like to do in a separate branch.  So maybe information like this could be added in the README.  I can do a pull request in the near future to document this.

But one simple request that you might consider would be to just add the tagging that other posters have agreed would be nice to have.   It can be as simple as:

```
publish:
  docker:
    storage_driver: overlay
    repo: plugins/drone-ecr
    tag:
      - latest
      - "0.4.$$BUILD_NUMBER"
    when:
      branch: release/0.4
      event: push
```

I don't know if you have your own versioning scheme, but if not, you could just use the `BUILD_NUMBER`.

**Comment by matteosuppo** [comment]
So... are there news on this matter? because with the latest version from 3 hours ago the problem occurres again.

**Comment by tboerger** [comment]
Have it worked for you before? The last change have been a documentation update

**Comment by matteosuppo** [comment]
No, it's the first time I tried, so it's probably something from before.

**Comment by tboerger** [comment]
@matteosuppo than please follow the troubleshooting section of the documentation, this error can apply for various reasons.
